server.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const mainRoutes = require('./routes/main');
+
+// connect to db
+mongoose.connect('mongodb://localhost/products', {
```

Comment:
In newer version of node localhost needs to be changed to 127.0.0.1 for it to work properly https://www.mongodb.com/community/forums/t/mongooseserverselectionerror-connect-econnrefused-127-0-0-1-27017/123421/9